pkg/controllers/bgp/bgpconf_controller.go

```diff
@@ -190,9 +222,27 @@ func (r *BgpConfReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	// The policy field must be indexed by the manager, so that we will be able to lookup BgpConf by a referenced ConfigMap name.
+	err := mgr.GetFieldIndexer().IndexField(context.Background(), &v1alpha2.BgpConf{}, policyField, func(rawObj runtime.Object) []string {
+		// Extract the ConfigMap name from the BgpConf Spec, if one is provided
+		bgpConf := rawObj.(*v1alpha2.BgpConf)
+		if bgpConf.Spec.Policy == "" {
+			return nil
+		}
+		return []string{bgpConf.Spec.Policy}
+	})
+	if err != nil {
+		return err
+	}
+
 	ctl, err := ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha2.BgpConf{}).
 		WithEventFilter(p).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			&handler.EnqueueRequestForObject{},
```

Comment:
I don't think "handler. Enqueuerequestforobject}" can be used here.

Please refer to  `handler.EnqueueRequestsFromMapFunc(r.findObjectsForConfigMap)` 

https://book.kubebuilder.io/reference/watching-resources/externally-managed.html